### ISSUE TITLE
Wrong closest endpoint assigned

### ISSUE THREAD

**Issue by Nestorbra** [nextdns/nextdns#187]
Dear all,

I have set your DNS (the ones assigned to the profile I created under my account) on a Router level and despite of my public IP is a German one (Google, Netflix, Whatismyip, DNS Leak Test, etc.. also detects it as a German one) I'm not getting assigned to the closest endpoint which should be the Frankfurt one. Instead of that I'm getting assigned to Milan if I connect through a PC (Linux OS) and to Brussels if I connect through a Smart Phone (Android). Attached screen shots for your reference.

Could you please advice?

Regards,

N F

![German_IP](https://user-images.githubusercontent.com/23265100/80705333-89f70580-8ae6-11ea-8ab4-a464a6c5c20c.jpg)
![Brussels_Endpoint](https://user-images.githubusercontent.com/23265100/80705337-8a8f9c00-8ae6-11ea-8311-263d65260bc9.png)
![Milan_Endpoint](https://user-images.githubusercontent.com/23265100/80705338-8b283280-8ae6-11ea-8430-91d914efe398.png)

**Comment by maghuro** [comment]
Are you using a VPN to assign a German IP?
If yes, you need to route Google range 216.239.32.0/21 through VPN (the router.nextdns.io method that is used to get the closest endpoint) and/or nextdns range 45.90.28.0/22 also thorough VPN so anycast can also get the nearest location to your VPN IP.

If you're NOT using vpn, the router.nextdns.io should get you the nearest PoP. Maybe it's failing and it's using anycast (and for some reason Milan has a better route to you than Frankfurt?)

I'm just throwing ideas, I struggled with the same question but I use vpn, and I figured out to solve it this way.

**Comment by mad-it** [comment]
I got scared for a minute there.... I ran the same test locally and got the following result:
![image](https://user-images.githubusercontent.com/1469047/80717316-16aabf00-8af9-11ea-8d0a-87791e22e810.png)

I was super confused that I did not see nextdns mentioned here.

After some digging I found this beautiful piece of log: 
`Switching endpoint: https://zepto-ams-1.edge.nextdns.io#78.142.193.36`

Basically the edge node is registered in Amsterdam but the originating country is UK?

**Comment by Nestorbra** [comment]
> Are you using a VPN to assign a German IP?
> If yes, you need to route Google range 216.239.32.0/21 through VPN (the router.nextdns.io method that is used to get the closest endpoint) and/or nextdns range 45.90.28.0/22 also thorough VPN so anycast can also get the nearest location to your VPN IP.
> 
> If you're NOT using vpn, the router.nextdns.io should get you the nearest PoP. Maybe it's failing and it's using anycast (and for some reason Milan has a better route to you than Frankfurt?)
> 
> I'm just throwing ideas, I struggled with the same question but I use vpn, and I figured out to solve it this way.

Yes, I'm connected to a German VPN server from my Asus Merlin Router (using its OpenVPN client). But the strange think is that I'm only getting this issue when I connect to the German server. If I choose another server (e.g. Spain or USA) it is routed to a proper endpoint. 

However, if I type "_https://router.nextdns.io_" on my browser I receive the following result:

**[{"hostname":"ipv4-vultr-fra-1.edge.nextdns.io","ip":"108.61.179.198"}]**

Does that means that NextDNS is doing its job properly and maybe DNS Leaks Test website is giving a wrong result?

**Comment by maghuro** [comment]
> > Are you using a VPN to assign a German IP?
> > If yes, you need to route Google range 216.239.32.0/21 through VPN (the router.nextdns.io method that is used to get the closest endpoint) and/or nextdns range 45.90.28.0/22 also thorough VPN so anycast can also get the nearest location to your VPN IP.
> > If you're NOT using vpn, the router.nextdns.io should get you the nearest PoP. Maybe it's failing and it's using anycast (and for some reason Milan has a better route to you than Frankfurt?)
> > I'm just throwing ideas, I struggled with the same question but I use vpn, and I figured out to solve it this way.
> 
> Yes, I'm connected to a German VPN server from my Asus Merlin Router (using its OpenVPN client). But the strange think is that I'm only getting this issue when I connect to the German server. If I choose another server (e.g. Spain or USA) it is routed to a proper endpoint.
> 
> However, if I type "_https://router.nextdns.io_" on my browser I receive the following result:
> 
> **[{"hostname":"ipv4-vultr-fra-1.edge.nextdns.io","ip":"108.61.179.198"}]**
> 
> Does that means that NextDNS is doing its job properly and maybe DNS Leaks Test website is giving a wrong result?

Dnsleaktest is failing the geolocation of some ip, I found that a couple of days ago. Right now I'm connected to Frankfurt pop with the IP 5.254.17.5. dnsleaktest says that IP is from UK, but if you search it on IP location.net for example, it says that it is from Germany, correctly 😊 Frankfurt more precisely

**Comment by Nestorbra** [comment]
> > > Are you using a VPN to assign a German IP?
> > > If yes, you need to route Google range 216.239.32.0/21 through VPN (the router.nextdns.io method that is used to get the closest endpoint) and/or nextdns range 45.90.28.0/22 also thorough VPN so anycast can also get the nearest location to your VPN IP.
> > > If you're NOT using vpn, the router.nextdns.io should get you the nearest PoP. Maybe it's failing and it's using anycast (and for some reason Milan has a better route to you than Frankfurt?)
> > > I'm just throwing ideas, I struggled with the same question but I use vpn, and I figured out to solve it this way.
> > 
> > 
> > Yes, I'm connected to a German VPN server from my Asus Merlin Router (using its OpenVPN client). But the strange think is that I'm only getting this issue when I connect to the German server. If I choose another server (e.g. Spain or USA) it is routed to a proper endpoint.
> > However, if I type "_https://router.nextdns.io_" on my browser I receive the following result:
> > **[{"hostname":"ipv4-vultr-fra-1.edge.nextdns.io","ip":"108.61.179.198"}]**
> > Does that means that NextDNS is doing its job properly and maybe DNS Leaks Test website is giving a wrong result?
> 
> Dnsleaktest is failing the geolocation of some ip, I found that a couple of days ago. Right now I'm connected to Frankfurt pop with the IP 5.254.17.5. dnsleaktest says that IP is from UK, but if you search it on IP location.net for example, it says that it is from Germany, correctly  Frankfurt more precisely

Ok, thanks for your replies. I'm going to close this thread.
